### DISCUSSION TITLE
Generate Migration from POCO

### DISCUSSION THREAD

**Discussion by Fungusware** [fluentmigrator/fluentmigrator#1503] [Ideas]
Would be awesome to allow a migration to be automatically generated based on the current class definite of a POCO.

Awesome work, I really really like this library.

**Comment by fubar-coder** [comment]
This would require storing the complete schema somewhere (either in the library or in the database) and would result in the same tooling like Entity Framework (Core). I'm not sure if it's possible for a project that hasn't nearly as much developers as EF (Core).

**Comment by Fungusware** [comment]
Fair call...

**Comment by PhenX** [comment]
It may be something possible with a project like https://github.com/martinjw/dbschemareader/ but woudl require a lot of work anyway.

Its seems to provide a system to generate SQL migration scripts after schema comparison, and DbContext and NHibernate code from a schema.
